// ng-jhipster/esm5/language/language.service.js
import { __decorate } from "tslib";
/*
 Copyright 2013-2020 the original author or authors from the JHipster project.

 This file is part of the JHipster project, see https://www.jhipster.tech/
 for more information.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
import { Injectable } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';
import { JhiConfigService } from '../config.service';
import * as i0 from "@angular/core";
import * as i1 from "@ngx-translate/core";
import * as i2 from "../config.service";
var JhiLanguageService = /** @class */ (function () {
    function JhiLanguageService(translateService, configService) {
        this.translateService = translateService;
        this.configService = configService;
        this.currentLang = 'en';
    }
    JhiLanguageService.prototype.init = function () {
        var config = this.configService.getConfig();
        this.currentLang = config.defaultI18nLang;
        this.translateService.setDefaultLang(this.currentLang);
        this.translateService.use(this.currentLang);
    };
    JhiLanguageService.prototype.changeLanguage = function (languageKey) {
        this.currentLang = languageKey;
        this.configService.CONFIG_OPTIONS.defaultI18nLang = languageKey;
        this.translateService.use(this.currentLang);
    };
    /**
     * @deprecated Will be removed when releasing generator-jhipster v7
     */
    JhiLanguageService.prototype.getCurrent = function () {
        return Promise.resolve(this.currentLang);
    };
    JhiLanguageService.prototype.getCurrentLanguage = function () {
        return this.currentLang;
    };
    JhiLanguageService.ctorParameters = function () { return [
        { type: TranslateService },
        { type: JhiConfigService }
    ]; };
    JhiLanguageService.ɵprov = i0.ɵɵdefineInjectable({ factory: function JhiLanguageService_Factory() { return new JhiLanguageService(i0.ɵɵinject(i1.TranslateService), i0.ɵɵinject(i2.JhiConfigService)); }, token: JhiLanguageService, providedIn: "root" });
    JhiLanguageService = __decorate([
        Injectable({
            providedIn: 'root'
        })
    ], JhiLanguageService);
    return JhiLanguageService;
}());
export { JhiLanguageService };
